crates/transaction-generator-lib/src/publishing/publish_util.rs

```diff
@@ -107,48 +98,62 @@ impl PackageHandler {
         );
         if self.is_simple {
             if version {
-                package.version(rng);
+                update_simple_move_version(package.get_mut_module("simple"));
             }
-            package.scramble(tracker.publishers[idx].fn_count, rng);
+            scramble_simple_move(
+                package.get_mut_module("simple"),
+                tracker.publishers[idx].fn_count,
+                rng,
+            );
         }
-        // info!("PACKAGE: {:#?}", package);
         package
     }
 }
 
 // Enum to define all packages known to the publisher code.
 #[derive(Clone, Debug)]
 pub enum Package {
-    Simple(Vec<(String, CompiledModule)>, PackageMetadata),
+    Simple(
+        Vec<(String, CompiledModule)>,
+        PackageMetadata,
+        Option<CompiledScript>,
+    ),
 }
 
 impl Package {
-    pub fn by_name(name: &str) -> Self {
+    pub fn by_name(pre_built: &'static dyn PreBuiltPackages, name: &str) -> Self {
         let (modules, metadata) = Self::load_package(
-            &raw_module_data::PACKAGE_TO_METADATA[name],
-            &raw_module_data::PACKAGE_TO_MODULES[name],
+            pre_built.package_metadata(name),
+            pre_built.package_modules(name),
         );
-        Self::Simple(modules, metadata)
+        let script = pre_built
+            .package_script(name)
+            .map(|code| CompiledScript::deserialize(code).expect("Script must deserialize"));
+        Self::Simple(modules, metadata, script)
     }
 
-    pub fn script(publisher: AccountAddress) -> TransactionPayload {
-        assert_ne!(publisher, AccountAddress::MAX_ADDRESS);
+    pub fn script(&self, publisher: AccountAddress) -> TransactionPayload {
+        match self {
+            Self::Simple(_, _, script_opt) => {
+                let mut script = script_opt
+                    .clone()
+                    .expect("Script not defined for wanted package");
+                assert_ne!(publisher, AccountAddress::MAX_ADDRESS);
```

Comment:
Why can't publisher be `AccountAddress::MAX_ADDRESS`? What is this address reserved for?

Reply:
from the line below - // Make sure dependencies link to published modules. Compiler V2 adds 0xf..ff so we need to skip it